//===-- emsg.proto - Steam CM message types -------------------------------===//
//
// Copyright (c) 2025 Nuclearist <nuclearist@teknology-hub.com>
// Part of tek-steamclient, under the GNU General Public License v3.0 or later
// See https://github.com/teknology-hub/tek-steamclient/blob/main/COPYING for
//    license information.
// SPDX-License-Identifier: GPL-3.0-or-later
//
//===----------------------------------------------------------------------===//
syntax = "proto3";

package tek.steamclient.cm;

option optimize_for = LITE_RUNTIME;

// Steam CM message types.
enum EMsg {
  EMSG_UNSPECIFIED = 0;
  EMSG_MULTI = 1;
  EMSG_PROTOBUF_WRAPPED = 2;
  EMSG_SERVICE_METHOD_RESPONSE = 147;
  EMSG_SERVICE_METHOD = 151;
  EMSG_CLIENT_HEARTBEAT = 703;
  EMSG_CLIENT_LOG_OFF = 706;
  EMSG_CLIENT_PING_RESPONSE = 712;
  EMSG_CLIENT_GAMES_PLAYED_NO_DATA_BLOB = 715;
  EMSG_CLIENT_GAMES_PLAYED = 742;
  EMSG_CLIENT_REGISTER_KEY = 743;
  EMSG_CLIENT_LOG_ON_RESPONSE = 751;
  EMSG_CLIENT_SET_HEARTBEAT_RATE = 755;
  EMSG_CLIENT_LOGGED_OFF = 757;
  EMSG_CLIENT_PING = 764;
  EMSG_CLIENT_ACCOUNT_INFO = 768;
  EMSG_CLIENT_LICENSE_LIST = 780;
  EMSG_CLIENT_GET_USER_STATS = 818;
  EMSG_CLIENT_GET_USER_STATS_RESPONSE = 819;
  EMSG_CLIENT_STORE_USER_STATS = 820;
  EMSG_CLIENT_STORE_USER_STATS_RESPONSE = 821;
  EMSG_CLIENT_GET_APP_OWNERSHIP_TICKET = 857;
  EMSG_CLIENT_GET_APP_OWNERSHIP_TICKET_RESPONSE = 858;
  EMSG_CLIENT_GAMES_PLAYED_WITH_DATA_BLOB = 5410;
  EMSG_CLIENT_GET_DEPOT_DECRYPTION_KEY = 5438;
  EMSG_CLIENT_GET_DEPOT_DECRYPTION_KEY_RESPONSE = 5439;
  EMSG_CLIENT_SERVER_UNAVAILABLE = 5500;
  EMSG_CLIENT_SERVERS_AVAILABLE = 5501;
  EMSG_CLIENT_LOG_ON = 5514;
  EMSG_CLIENT_REQUEST_ENCRYPTED_APP_TICKET = 5526;
  EMSG_CLIENT_REQUEST_ENCRYPTED_APP_TICKET_RESPONSE = 5527;
  EMSG_CLIENT_PICS_CHANGES_SINCE_REQUEST = 8901;
  EMSG_CLIENT_PICS_CHANGES_SINCE_RESPONSE = 8902;
  EMSG_CLIENT_PICS_PRODUCT_INFO_REQUEST = 8903;
  EMSG_CLIENT_PICS_PRODUCT_INFO_RESPONSE = 8904;
  EMSG_CLIENT_PICS_ACCESS_TOKEN_REQUEST = 8905;
  EMSG_CLIENT_PICS_ACCESS_TOKEN_RESPONSE = 8906;
  EMSG_SERVICE_METHOD_CALL_FROM_CLIENT_NON_AUTHED = 9804;
  EMSG_CLIENT_HELLO = 9805;
};
